contrib/epee/include/net/levin_protocol_handler_async.h

```diff
@@ -595,9 +626,15 @@ class async_protocol_handler
                             << ", ver=" << head.m_protocol_version);
 
     uint64_t ticks_start = misc_utils::get_tick_count();
+    size_t prev_size = 0;
 
     while(!boost::interprocess::ipcdetail::atomic_read32(&m_invoke_buf_ready) && !m_deletion_initiated && !m_protocol_released)
     {
+      if(m_cache_in_buffer.size() - prev_size >= MIN_BYTES_WANTED)
+      {
+        prev_size = m_cache_in_buffer.size();
+        ticks_start = misc_utils::get_tick_count();
+      }
```

Comment:
That code seems to rely on another function to full m_cache_in_buffer. I see that's in handle_recv. If this is another thread, this is racy since it can also erase some data from m_cache_in_buffer.  But it could also be called from m_pservice_endpoint->call_run_once_service_io() below, which would be OK. I don't know boost enough to know without hunting,so I'll ask in case you know.